cuckoo/common/abstracts.py

```diff
@@ -38,6 +38,10 @@ def __init__(self):
         self.guest_manager = None
         self.options = None
 
+    @classmethod
```

Comment:
I went with using a "classmethod" so that the new update_config method would not have to create an instance of all the various plugins. I do not imagine plugin developers will make use of the fact that it is a "classmethod" but this seemed better than creating a new instance of all the plugins in update_config(). Also the change does not require plugin developers to implement "get_config" and they can continue to modify config.py if they wish too.